### PR TITLE
Add options to deploy the proxy as DaemonSet

### LINKED ISSUE #84: Deploy as DaemonSet with Helm
In some deployments, it would be useful to deploy the proxy as daemonset on the control plane nodes.
Update the Helm chart with this option.

### PR DESCRIPTION
**Pull request by bsctl**
close #84

### CODE REVIEW

File: charts/capsule-proxy/templates/daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      {{- with .Values.affinity }}
+      affinity:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      {{- with .Values.tolerations }}
+      tolerations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      restartPolicy: {{ .Values.restartPolicy }}
+{{- end }}
```
Comment (prometherion) [inline charts/capsule-proxy/templates/daemonset.yaml:102]: Missing new line at the end of the file.

File: charts/capsule-proxy/templates/deployment.yaml [hunk 2]
```diff
@@ -6,7 +7,7 @@ metadata:
     {{- include "capsule-proxy.labels" . | nindent 4 }}
 spec:
   strategy:
-    type: Recreate
+    type: RollingUpdate
```
Comment (prometherion) [inline charts/capsule-proxy/templates/deployment.yaml:10]: Nice catch

File: charts/capsule-proxy/templates/deployment.yaml [hunk 3]
```diff
@@ -99,3 +94,5 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+      restartPolicy: {{ .Values.restartPolicy }}
+{{- end }}
```
Comment (prometherion) [inline charts/capsule-proxy/templates/deployment.yaml:98]: Missing new line at the end of the file.

File: charts/capsule-proxy/values.yaml [hunk 4]
```diff
@@ -2,17 +2,15 @@
...
+kind: Deployment # Deployment | DaemonSet
 image:
   repository: quay.io/clastix/capsule-proxy
   pullPolicy: IfNotPresent
   tag: ""
+imagePullSecrets: []
 options:
   listeningPort: 9001
-  logLevel: '5'
+  logLevel: '4'
```
Comment (prometherion) [inline charts/capsule-proxy/values.yaml:13]: Why decrease from 5?

**Review by prometherion — CHANGES_REQUESTED** [review]
Minor things, but overall LGTM.

File: charts/capsule-proxy/values.yaml (hunk 4 above)
Comment (bsctl) [reply charts/capsule-proxy/values.yaml:13]: Make it consistent with capsule log verbosity set to 4 as default

**Review by prometherion — APPROVED** [review]
LGTM.

@MaxFedotov is it the same for you?

**Comment by MaxFedotov** [comment]
lgtm too

File: charts/capsule-proxy/templates/deployment.yaml [hunk 5]
```diff
@@ -22,17 +23,14 @@ spec:
       labels:
         {{- include "capsule-proxy.selectorLabels" . | nindent 8 }}
     spec:
-      {{- if .Values.hostNetwork }}
```
Comment (prometherion) [inline charts/capsule-proxy/templates/deployment.yaml:25]: The removal of the `hostNetwork` is letting fail the e2e case, it should be put back.

**Review by prometherion — CHANGES_REQUESTED** [review]

File: charts/capsule-proxy/templates/deployment.yaml (hunk 5 above)
Comment (bsctl) [reply charts/capsule-proxy/templates/deployment.yaml:25]: @prometherion run e2e test as `DaemonSet` with `hostNetwork=true` now succeeds.

**Review by prometherion — APPROVED** [review]
LGTM

**Pull request merged**